### ISSUE TITLE
NaN in loss maybe due to ...

### ISSUE THREAD

**Issue by thegodone** [aimat-lab/gcnn_keras#42]
/Users/itetko/anaconda3/lib/python3.8/site-packages/kgcnn-1.1.2-py3.8.egg/kgcnn/utils/adj.py:67: RuntimeWarning: divide by zero encountered in power
  d_ii = np.power(d_row, -0.5).flatten()
/Users/itetko/anaconda3/lib/python3.8/site-packages/kgcnn-1.1.2-py3.8.egg/kgcnn/utils/adj.py:68: RuntimeWarning: divide by zero encountered in power
  d_jj = np.power(d_col, -0.5).flatten()

I see this error coming (I find it in DMPNN)

**Comment by PatReis** [comment]
Okay, there should not be a warning any more but also no nan or inf appear in the edge weights.
Do you have problems with NaN in loss? I recently also had problems with PAiNN being unstable on tf-version 2.7 but the exact same dataset and network was fine on 2.4 (which confused me a lot.)
But I also heard colleagues that they had problems with training PAiNN to be stable with PyTorch...
That is why I want to see if I can properly place Normalization layers in PAiNN.

**Comment by thegodone** [comment]
it's not always NaN in loss it may appends. I don't have tf 2.7 yes. I also see that Keras is much better than Pytorch in converging. I guess normalization can solve it or clip the gradient to avoid explosion
